### PR TITLE
mrs: handle empty quarantine

### PR DESCRIPTION
**Pull request by brooksdavis**
Add a test for calling malloc_revoke() with an empty quarantine in mrs and fix a crash when that happens.

Fixes: #1875

### CODE REVIEW

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 1]
```diff
@@ -719,6 +719,9 @@ static inline void quarantine_flush(struct mrs_quarantine *quarantine) {
 	}
 #endif /* !JUST_QUARANTINE && !JUST_PAINT_BITMAP */
 
+	if (quarantine->list == NULL)
+		return;
```
Comment (jrtc27) [inline lib/libc/stdlib/malloc/mrs/mrs.c:723]: This loses the mrs_debug_printf at the end of the function. Can we just guard the problematic few lines involving prev with a null check instead?

File: bin/cheribsdtest/cheribsdtest_malloc.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2023 (holder)
```
Comment (jrtc27) [inline bin/cheribsdtest/cheribsdtest_malloc.c:4]: (holder)

File: bin/cheribsdtest/Makefile.cheribsdtest [hunk 3]
```diff
@@ -30,6 +30,7 @@ SRCS+=	cheribsdtest_bounds_globals.c					\
 .if ${MACHINE_ABI:Mpurecap}
 SRCS+=	cheribsdtest_bounds_subobject.c					\
 	cheribsdtest_bounds_varargs.c					\
+	cheribsdtest_malloc.c						\
```
Comment (jrtc27) [inline bin/cheribsdtest/Makefile.cheribsdtest:33]: This won't work without MRS?

Comment (brooksdavis) [reply bin/cheribsdtest/Makefile.cheribsdtest:33]: There's a stub malloc_is_revoking() so they will skip with non-mrs allocators. While we could disable it to avoid skipped tests, I hope to move away from mrs at some point in the not to distant future.

Comment (jrtc27) [reply bin/cheribsdtest/Makefile.cheribsdtest:33]: Ok, that's fine then
